### PR TITLE
Update to latest LUIS SDK, update entity recognizers and consolidate versions.

### PR DESCRIPTION
**Pull request by chrimc62**

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs [hunk 1]
```diff
@@ -15,8 +15,8 @@ public class LuisApplication
         /// </summary>
         /// <param name="applicationId">LUIS _application ID.</param>
         /// <param name="endpointKey">LUIS subscription or endpoint key.</param>
-        /// <param name="azureRegion">Azure region with endpoint.</param>
-        public LuisApplication(string applicationId, string endpointKey, string azureRegion)
+        /// <param name="endpoint">LUIS endpoint to use.</param>
+        public LuisApplication(string applicationId, string endpointKey, string endpoint)
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs]: endpointKey is not being argument / null checked checked.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs [hunk 2]
```diff
@@ -61,11 +55,11 @@ public LuisApplication(string applicationId, string endpointKey, string azureReg
         public string EndpointKey { get; set; }
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs]: Given that this is set in the constructor, consider making this "get-only" and removing the setter. No reason to change this once the object is constructed.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs [hunk 3]
```diff
@@ -61,11 +55,11 @@ public LuisApplication(string applicationId, string endpointKey, string azureReg
...
         /// <summary>
-        /// Gets or sets Azure Region where endpoint is located.
+        /// Gets or sets LUIS endpoint.
         /// </summary>
         /// <value>
-        /// Azure Region where endpoint is located.
+        /// LUIS endpoint where application is hosted.
         /// </value>
-        public string AzureRegion { get; set; }
+        public string Endpoint { get; set; }
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs]: Given that this is set in the constructor, consider making this "get-only" and removing the setter. No reason to change this once the object is constructed.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs [hunk 4]
```diff
@@ -31,7 +31,7 @@ public class LuisRecognizer : IRecognizer
         /// </summary>
         public const string LuisTraceLabel = "Luis Trace";
         private const string _metadataKey = "$instance";
-        private readonly LuisRuntimeAPI _runtime;
+        private readonly ILUISRuntimeClient _runtime;
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: Casing. In many examples ("LuisApplication", "LuisTraceLabel", "LuisPredictionOptions") LUIS is lower case. In this case it's LUIS. 

Please choose one. :)

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs [hunk 5]
```diff
@@ -45,9 +45,9 @@ public class LuisRecognizer : IRecognizer
         /// <param name="clientHandler">Custom handler for LUIS API calls.</param>
         public LuisRecognizer(LuisApplication application, LuisPredictionOptions predictionOptions = null, bool includeApiResults = false, HttpClientHandler clientHandler = null)
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: "LUIS _application" on line 42 has an extra space in it.

Comment (cleemullins) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: Line 42. Add a <see> tag to LuisApplication. 
Line 43. Add a <see> tag to LuisPredictionOptions.
Line 43. Add (Optional) to the start of the text. 
Line 44. Add (Optional) to includeApiResults
Line 45. Add (Optional) to the clientHandler.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs [hunk 6]
```diff
@@ -45,9 +45,9 @@ public class LuisRecognizer : IRecognizer
...
         {
-            _runtime = new LuisRuntimeAPI(new ApiKeyServiceClientCredentials(application.EndpointKey), clientHandler)
+            _runtime = new LUISRuntimeClient(new ApiKeyServiceClientCredentials(application.EndpointKey), clientHandler)
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: Null check application.
Null check application.EndpointKey.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj [hunk 7]
```diff
@@ -62,10 +62,10 @@
 	</PropertyGroup>
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj]: The 3 CodeAnalysisRuleSet elements above, in the conditional Elements, should be removed. The ruleset shouldn't be conditional. A non-conditional once is already present on line 59 which is perfect.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj [hunk 8]
```diff
@@ -62,10 +62,10 @@
...
 
 	<ItemGroup>
-		<PackageReference Include="Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime" Version="0.9.0-preview" />
+		<PackageReference Include="Microsoft.Azure.CognitiveServices.Language.LUIS.Runtime" Version="2.0.0" />
```
Comment (cleemullins) [inline libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj]: Casing on LUIS. The same casing should be used everywhere.

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisApplicationTests.cs [hunk 9]
```diff
@@ -10,20 +10,23 @@ namespace Microsoft.Bot.Builder.AI.Luis.Tests
     [TestClass]
     public class LuisApplicationTests
     {
+        private const string Endpoint = "https://westus.api.cognitive.microsoft.com/luis/v2.0/";
+
         [TestMethod]
         public void LuisApplication_Construction()
         {
             var model = GetValidModel();
+            var badEndpoint = "abc";
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisApplicationTests.cs]: const string

File: tests/Microsoft.Bot.Builder.Ai.LUIS.Tests/TestData/review.cmd [hunk 10]
```diff
@@ -0,0 +1,21 @@
+@echo off
```
Comment (cleemullins) [inline tests/Microsoft.Bot.Builder.Ai.LUIS.Tests/TestData/review.cmd]: Please add some comments into this file describing what it does. I've no idea!

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs (hunk 4 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: This is from the underlying LUIS SDK--where they have changed the names.  I'm happy to go to all LUIS--that was my preference but did not seem to be the naming guidelines we were following in the V4 SDK.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs (hunk 1 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs]: It is null whitespace and URI checked.  Not sure what more you could do...

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs (hunk 2 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisApplication.cs]: Done.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs (hunk 4 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: I'm going to leave this as Luis in all of SDK V4.

**Comment by daveta** [comment]
@chrimc62 : Not sure if @cleemullins issues addressed, but would love to get this checked in so I get some more samples in..

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs (hunk 5 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: Done.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs (hunk 6 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs]: Done

File: tests/Microsoft.Bot.Builder.Ai.LUIS.Tests/TestData/review.cmd (hunk 10 above)
Comment (chrimc62) [reply tests/Microsoft.Bot.Builder.Ai.LUIS.Tests/TestData/review.cmd]: Done

File: tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisApplicationTests.cs (hunk 9 above)
Comment (chrimc62) [reply tests/Microsoft.Bot.Builder.AI.LUIS.Tests/LuisApplicationTests.cs]: Removed.

File: libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj (hunk 7 above)
Comment (chrimc62) [reply libraries/Microsoft.Bot.Builder.AI.LUIS/Microsoft.Bot.Builder.AI.LUIS.csproj]: Done

**Pull request merged**